### PR TITLE
LOG4J2-3080 Use SimpleMessage in Log4j 1 Category whenever possible

### PR DESCRIPTION
**Pull request by vy**

### CODE REVIEW

**Review by carterkozak — APPROVED** [review]
lgtm

File: log4j-1.2-api/src/main/java/org/apache/log4j/Category.java [hunk 1]
```diff
@@ -507,11 +508,27 @@ public void log(final String fqcn, final Priority priority, final Object message
...
+            if (message instanceof String) {
+                msg = new SimpleMessage((String) message);
+            }
+            // SimpleMessage treats String and CharSequence differently, hence
+            // this else-if block.
+            else if (message instanceof CharSequence) {
+                msg = new SimpleMessage((CharSequence) message);
+            } else if (message instanceof Map) {
+                @SuppressWarnings("unchecked")
+                final Map<String, ?> map = (Map<String, ?>) message;
```
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/Category.java:527]: Is the map local var really needed?

**Review by rgoers — APPROVED** [review]
Other than what Gary commented on this looks fine to me.

File: log4j-1.2-api/src/main/java/org/apache/log4j/Category.java (hunk 1 above)
Comment (vy) [reply log4j-1.2-api/src/main/java/org/apache/log4j/Category.java:527]: It is there to suppress `javac` warnings via `@SuppressWarnings("unchecked")`.

**Comment by vy** [comment]
@garydgregory @carterkozak @rgoers Thanks for the review. Will merge this and port to `master`.

**Pull request merged**